### PR TITLE
feat(import-target): Add resolution error reason

### LINKED ISSUE #182: Bug: `import-target` mutes resolution errors
Currently in [`lib/util/import-target.js:273`](https://github.com/eslint-community/eslint-plugin-n/blob/master/lib/util/import-target.js#L273) we skip all errors as if it errors the package is not found.
However, we need to be able to display some of the errors as not all not found errors are equal!

---

An example:

My file:
```js
// index.js
import 'missing';
import 'misconfigured';
```

```json
{
  "name": "demo",
  "type": "module",
  "dependencies": {
    "misconfigured": "*"
  }
}
```

When resolving `missing` the error is:
> `Can't resolve 'soup' in '/home/scagood/issue-demo'`

When resolving `misconfigured` the error is:
> `Default condition should be last one`

In `node_modules/misconfigured/package.json` the following mistake was made:
```json
{
  "name": "misconfigured",
  "exports": {
    ".": {
      "default": "./build/index.js",
      "types": "./src/index.ts"
    }
  },
  ...
}
```

In order to resolve this, the `exports` should become:
```json
{
  "name": "misconfigured",
  "exports": {
    ".": {
      "types": "./src/index.ts",
      "default": "./build/index.js"
    }
  },
  ...
}
```

---

This is because `enhanced-resolve` explicitly throws an error here:
https://github.com/webpack/enhanced-resolve/blob/main/lib/util/entrypoints.js#L475

I am not 100% sure why... but it is an error that we need to be able to display!

### PR DESCRIPTION
**Pull request by aladdin-add**
Fixed test cases (based on https://github.com/eslint-community/eslint-plugin-n/pull/231).

fixes #182

### CODE REVIEW

File: tests/eslint-rule-tester.js [hunk 1]
```diff
@@ -8,12 +8,17 @@ const { RuleTester } = require("eslint")
...
 
 // greater than or equal to ESLint v9
 exports.gteEslintV9 = semverSatisfies(eslintVersion, ">=9", {
     includePrerelease: true,
 })
 
+const platform = os.platform()
+exports.isCaseSensitiveFileSystem =
+    platform === "linux" || platform === "freebsd" || platform === "openbsd"
+
```
Comment (scagood) [inline tests/eslint-rule-tester.js:21]: Can we make this in a different file? I don't think it belongs with the rule tester

File: tests/eslint-rule-tester.js [hunk 2]
```diff
@@ -8,12 +8,17 @@ const { RuleTester } = require("eslint")
 const { FlatRuleTester } = require("eslint/use-at-your-own-risk")
```
Comment (scagood) [inline tests/eslint-rule-tester.js:8]: I am starting to think we should make these patches for different eslint versions into a package

**Review by scagood — APPROVED** [review]
A couple of small non blocking comments, but in general I like it! Thank you

File: tests/eslint-rule-tester.js (hunk 1 above)
Comment (aladdin-add) [reply tests/eslint-rule-tester.js:21]: right, renamed to `test-helpers`. :)

**Pull request merged**